### DISCUSSION TITLE
Can I use .obj files?

### DISCUSSION THREAD

**Discussion by andytriboletti** [sceneview/sceneview#448] [Q&A]
How can I display .obj files with .mtl and texture, such as those exported by threestudio? Do I first have to convert them on command line or is there some way to display them without conversion like in Swift?

**Comment by v01dn3ph1l1m** [comment]
The names of the obj and mtl files should be the same. Load the obj file like you would a glb or glTF file. Obj is supported by filament

**Reply by Danwolve98** [reply]
Im trying do the same, but when i try to load a obj file it crashes, could u show us a example?

**Comment by ThomasGorisse** [comment]
SceneView uses Filament which supports **glTF/GLB** format. OBJ files are not directly supported.

To use OBJ files, convert them to GLB first:
- **Blender**: Import OBJ, export as GLB
- **Online**: Use tools like [gltf.report](https://gltf.report) or Khronos glTF tools
- **CLI**: Use `obj2gltf` from Cesium (`npm install -g obj2gltf`)

GLB is recommended because it's a single binary file that packages meshes, materials, textures, and animations together, and it's more efficient to load at runtime.
